### ISSUE TITLE
Length optimization on message syntax

### ISSUE THREAD

**Issue by jordicenzano** [kixelated/warp-draft#206]
Since message (6.  Messages) will be the main piece of info of MOQ I think we should try to minimize its size. Assuming most of the messages send to the wire will be objects inside a quic stream, the length should be NOT needed. Could we use 1 bit in "type" to indicate if length is present, we could save some bytes without overcomplicating the protocol

**Comment by VMatrix1900** [comment]
The length of a QUIC stream is only known after the FIN bit is received(ie the end of QUIC stream). Putting the length in the message header can ensure the size of stream is known at the beginning of the stream.

**Comment by kixelated** [comment]
The size of an OBJECT is a varint so that's a single byte when 0. Even in the situation where you're sending an OBJECT per frame, that's what like 100 OBJECTs per second?. That's 100B/s while the media could be 375,000B/s (3Mb/s).

But yeah, we can optimize when the draft is closer to completion and thinks are more concrete. I like how QUIC STREAM frames are encoded, where the frame type indicates if fields are present (ex. length).

**Comment by ianswett** [comment]
This issue is related to the question of what types of lengths we'll support, and whether we'll support those that go to the end of a quic stream (aka infinite length).

"unknown" payloads where they're a delimiter of some sort had no support, because you have to do extra parsing.

This issue is about two types: Object and Object + Length, which has support from the WG.

Martin to roll into PR #255

**Comment by martinduke** [comment]
From Cambridge: Martin will put this in a separate PR once #255 lands
